src/ussd_registration.js

```diff
@@ -0,0 +1,343 @@
+                    }
+                },
+                next: 'state_msg_language'
+            });
+        });
+
+        // ChoiceState st-07
+        self.add('state_msg_language', function(name) {
+            return new ChoiceState(name, {
+                question: $(questions[name]),
+                error: $(get_error_text(name)),
+                choices: [
+                    new Choice('english', $('English')),
+                    new Choice('hausa', $('Hausa')),
+                    new Choice('igbo', $('Igbo'))
+                ],
+                next: 'state_msg_call_or_text'
+            });
+        });
+
+        // ChoiceState st-08
+        self.add('state_msg_call_or_text', function(name) {
+            return new ChoiceState(name, {
+                question: $(questions[name]),
+                error: $(get_error_text(name)),
+                choices: [
+                    new Choice('voice_calls', $('Voice calls')),
+                    new Choice('text_smss', $('Text SMSs'))
+                ],
+                next: function(choice) {
+                    return choice.value === 'voice_calls'
+                        ? 'state_receive_calls_days'
+                        : 'state_end_thank_you_texts';
+                }
+            });
+        });
+
+        // ChoiceState st-09
+        self.add('state_receive_calls_days', function(name) {
+            return new ChoiceState(name, {
+                question: $(questions[name]),
+                error: $(get_error_text(name)),
+                choices: [
+                    new Choice('monday_and_wednesday', $('Monday and Wednesday')),
+                    new Choice('tuesday_and_thursday', $('Tuesday and Thursday'))
+                ],
+                next: 'state_receive_calls_time'
+            });
+        });
+
+        // ChoiceState st-10
+        self.add('state_receive_calls_time', function(name) {
+            return new ChoiceState(name, {
+                question: $(questions[name]),
+                error: $(get_error_text(name)),
+                choices: [
+                    new Choice('between_9_11am', $('Between 9-11am')),
+                    new Choice('between_2_5pm', $('Between 2-5pm'))
+                ],
+                next: 'state_end_thank_you_calls'
+            });
+        });
+
+        // EndState st-11
+        self.add('state_end_thank_you_calls', function(name) {
+            return new EndState(name, {
+                text: $(questions[name]),
+                next: 'state_start'
+            });
+        });
+
+        // PaginatedChoiceState st-12
+        self.add('state_baby_birth_date', function(name) {
+            var today = go.utils.get_today(self.im.config);
+            var start_month = today.month();
+            return new PaginatedChoiceState(name, {
+                question: $(questions[name]),
+                error: $(get_error_text(name)),
+                characters_per_page: 160,
```

Comment:
Let's set this to 182 for now to line up with the test harness. No info yet on what it needs to be.